kornia/augmentation/base.py

```diff
@@ -50,6 +52,12 @@ def __infer_input__(
     ) -> torch.Tensor:
         return input
 
+    def __ensure_same_batching_output__(self, output: Union[torch.Tensor,
```

Comment:
This check shall happen after ```__infer_input__ ```, to validate the input value/tuple.

For the naming, this function performed a checking than ensuring. Maybe ```__check_batching__``` or ```__check_input__``` or something similar.